source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -70,6 +93,27 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap&, bool) {
 
   config->stats().enabled_.inc();
 
+  std::vector<Envoy::RateLimit::Descriptor> descriptors;
+  getRouteSpecificDescriptors(descriptors, headers);
```

Comment:
It would be nice to somehow gate this code entirely on whether there are any descriptors configured at all.

Reply:
Done.